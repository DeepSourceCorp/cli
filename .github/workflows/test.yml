name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
