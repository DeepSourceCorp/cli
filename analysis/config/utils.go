package config

import (
	"log"
	"os"
	"path"
	"path/filepath"
	"strings"

	"github.com/bmatcuk/doublestar/v4"
)

/* Walks the `CODE_PATH` directory and returns all the files other than the ones present
 * in the .git folder for analysis in the form of a string array */
func readAllFiles(codePath string) ([]string, error) {
	fileCount := 0

	allFiles := make([]string, 0)

	err := filepath.Walk(codePath,
		func(path string, fileInfo os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			fileCount++

			/* Check the following before appending to the list of files:
			 * - Should not be a directory
			 * - The walked file should not be present in .git folder
			 * TODO: Check why fileCount != 1 condition is used here */
			if !fileInfo.IsDir() || !strings.HasPrefix(path, filepath.Join(codePath, ".git")) && fileCount != 1 {
				allFiles = append(allFiles, path)
			}
			return nil
		})
	if err != nil {
		return allFiles, err
	}
	return allFiles, nil
}

// Returns the slice of files matching certain glob patterns
func (r *AnalysisRun) getMatchingFiles(patterns []string) ([]string, error) {
	matchedFiles := make([]string, 0)

	// Return all the files if no exclude_patterns are configured
	if len(patterns) == 0 {
		return matchedFiles, nil
	}

	for _, file := range r.AnalysisFiles {
		for i := range patterns {
			match, err := doublestar.Match(path.Join(r.CodePath, patterns[i]), file)
			if err != nil {
				log.Printf("Failed to match the file %s with pattern %s\n", file, patterns[i])
			}
			if match {
				matchedFiles = append(matchedFiles, file)
				break
			}
		}
	}
	return matchedFiles, nil
}

// Filters the analysis files and removes the files matching the exclude_patterns from them
// TODO: Improve the logic here
func (r *AnalysisRun) filterAnalysisFiles() {
	filteredFiles := []string{}
	for _, file := range r.AnalysisFiles {
		for _, excludedFile := range r.ExcludedFiles {
			if file == excludedFile {
				break
			}
			filteredFiles = append(filteredFiles, file)
		}
	}
	r.AnalysisFiles = filteredFiles
}
