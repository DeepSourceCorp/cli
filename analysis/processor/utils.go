package processor

import (
	"sort"

	"github.com/deepsourcelabs/cli/types"
)

// sortIssuesByFile sorts the issues in an alphabetical order according to the filenames
// where they got reported.
func (p *ProcessAnalysisResults) sortIssuesByFile() {
	sort.Slice(p.AnalysisResult.Issues, func(i, j int) bool {
		return p.AnalysisResult.Issues[i].Location.Path < p.AnalysisResult.Issues[j].Location.Path
	})
}

// Prepare a map with unique filenames as key and the issue range for each file as value
// This is done to ensure fewer loops when processing issues.
type IssueRange struct {
	Filename   string // The file which has the issues
	BeginIndex int    // Array index in report.Issues where the particular issue starts
	EndIndex   int    // Array index in report.Issues where the particular issue ends
}

// GenerateIssueRangeSlice generates an array containing the issue ranges with respect to files
// that helps us to go through them and map them to the files where they got reported instead
// of opening the file for each of them. The generated index looks like this:

// [{analyzer.go 0 0} {autofix_patch.go 1 1} {difftool.go 2 2} {patch.patch 3 5} {proc_skip_cq_test.go 6 6}]

// Here, the first field if filename and the second and third fields are the index range in which the issues reported
// in these files lie in the sorted AnalyzerReport slice.
func createIssueFileRange(report types.AnalysisResult) []IssueRange {
	fileCount := 0 // for 1 file, 0 based indexing
	issuesRange := []IssueRange{}
	prevFilename := report.Issues[0].Location.Path

	issueRange := IssueRange{
		BeginIndex: 0,
		EndIndex:   len(report.Issues) - 1,
		Filename:   prevFilename,
	}
	issuesRange = append(issuesRange, issueRange)

	// Iterating over the issues and creating an array containing issues with index data about
	// the files in which those issues are present
	for i := 1; i < len(report.Issues); i++ {
		issue := report.Issues[i]
		currentFilename := issue.Location.Path

		// TODO: Check when this condition is implied
		if issue.Location.Position.End.Line == -1 {
			issue.Location.Position.End.Line = issue.Location.Position.Begin.Line
		}

		if currentFilename != prevFilename {
			fileCount++

			issueRange = issuesRange[fileCount-1]
			issueRange.EndIndex = i - 1
			issuesRange[fileCount-1] = issueRange

			// Create for the new file
			issueRange = IssueRange{
				Filename:   currentFilename,
				BeginIndex: i,
				EndIndex:   len(report.Issues) - 1,
			}

			issuesRange = append(issuesRange, issueRange)
			prevFilename = currentFilename
		}
	}
	return issuesRange
}

// formatLSPResultsToDefault converts the LSP based analysis results into the default format supported by DeepSource.
func (p *ProcessAnalysisResults) formatLSPResultsToDefault() {
	p.AnalysisResult.IsPassed = p.Report.IsPassed
	p.AnalysisResult.Metrics = append(p.Report.Metrics, p.AnalysisResult.Metrics...)
	p.AnalysisResult.Errors = append(p.Report.Errors, p.AnalysisResult.Errors...)

	// Appending the issues to the default format of Analysis report
	for _, issue := range p.Report.Issues {
		analysisIssue := types.Issue{
			IssueCode: issue.Code,
			IssueText: issue.Message,
			Location: types.Location{
				Path: issue.RelatedInformation[0].Location.URI,
				Position: types.Position{
					Begin: types.Coordinate{
						Line:   issue.Range.Start.Line,
						Column: issue.Range.Start.Character,
					},
					End: types.Coordinate{
						Line:   issue.Range.End.Line,
						Column: issue.Range.End.Character,
					},
				},
			},
		}
		p.AnalysisResult.Issues = append(p.AnalysisResult.Issues, analysisIssue)
	}
}
