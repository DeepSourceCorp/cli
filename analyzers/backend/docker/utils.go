package docker

import (
	"bufio"
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"strings"
)

type ErrorLine struct {
	Error       string      `json:"error"`
	ErrorDetail ErrorDetail `json:"errorDetail"`
}

type ErrorDetail struct {
	Message string `json:"message"`
}

type DockerBuildResponse struct {
	Stream string `json:"stream"`
}

func CheckBuildResponse(rd io.Reader, showAllLogs bool) error {
	var lastLine []byte
	count := 0
	var currentStream string

	scanner := bufio.NewScanner(rd)
	for scanner.Scan() {
		lastLine = scanner.Bytes()
		d := &DockerBuildResponse{}
		err := json.Unmarshal(lastLine, d)
		if err != nil {
			return err
		}
		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
			continue
		}
		currentStream = strings.TrimSuffix(d.Stream, "\n")
		if showAllLogs {
			fmt.Println(currentStream)
		}
		count++
	}

	errLine := &ErrorLine{}
	json.Unmarshal([]byte(lastLine), errLine)
	if errLine.Error != "" {
		return errors.New(errLine.Error)
	}
	return scanner.Err()
}


