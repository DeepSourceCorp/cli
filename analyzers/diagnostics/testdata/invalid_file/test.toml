name = "2do-Checker"
