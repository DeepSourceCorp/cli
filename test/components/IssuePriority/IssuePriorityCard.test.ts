import { fireEvent, render } from '@testing-library/vue'
import { IssuePriorityCard } from '~/components/IssuePriority'
import { BaseCard } from '~/components/History'
import { cartesian, generateBooleanProps, generateStringProps } from '~/test/utils'
import VTooltip from 'floating-vue'

test('renders IssuePriorityCard with all prop options', () => {
  const baseProps = {
    title: 'Assert statement used outside of tests',
    shortcode: 'BAN-B101',
    issueType: 'security',
    analyzer: {
      name: 'Python',
      logo: 'analyzer_logos/python.svg',
      shortcode: 'python'
    }
  }

  const priorityOptions = generateStringProps('priority', ['High', 'Medium', 'Low', 'noop'], false)
  const canChangePriorityOptions = generateBooleanProps('canChangePriority', false)

  cartesian(priorityOptions, canChangePriorityOptions).forEach((propCombination) => {
    const props = {
      ...baseProps,
      ...propCombination
    }

    const { html } = render(
      IssuePriorityCard,
      {
        props,
        stubs: {
          ZIcon: true,
          ZButton: true,
          ZMenuSection: true,
          ZMenuItem: true,
          AnalyzerLogo: true,
          BaseCard
        }
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot(JSON.stringify(props))
  })
})

test('Emits delete event', async () => {
  const shortcode = 'BAN-B101'
  const { getByTestId, emitted } = render(
    IssuePriorityCard,
    {
      props: {
        title: 'Assert statement used outside of tests',
        shortcode: 'BAN-B101',
        issueType: 'security',
        analyzer: {
          name: 'Python',
          logo: 'analyzer_logos/python.svg',
          shortcode: 'python'
        },
        priority: 'High',
        canChangePriority: true
      },
      stubs: {
        ZIcon: true,
        ZMenuSection: true,
        ZMenuItem: true,
        AnalyzerLogo: true,
        BaseCard
      }
    },
    (vue) => {
      vue.use(VTooltip)
    }
  )

  const button = getByTestId('unset-issue-priority')

  await fireEvent(button, new Event('click'))
  expect(emitted()).toHaveProperty('priority-unset')
  expect(emitted()['priority-unset'][0][0]).toBe(shortcode)
})
