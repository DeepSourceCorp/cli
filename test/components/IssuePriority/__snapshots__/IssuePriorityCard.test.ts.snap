// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":false,"priority":"High"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\">
        <div class=\\"flex items-center gap-x-1 uppercase bg-ink-200 rounded-full px-1.5 py-1 cursor\\">
          <zicon-stub icon=\\"ellipsis-small\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"-mx-0.5\\"></zicon-stub> <span class=\\"text-xxs text-vanilla-400 tracking-wide font-medium\\">

  </span>
        </div>
        <div class=\\"flex items-center gap-x-1.5\\">
          <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
        </div>
        <div class=\\"flex items-center space-x-1.5\\">
          <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
        </div>
      </div>
    </div>
    <div class=\\"hidden w-full md:block md:w-1/5\\"></div>
  </div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":false,"priority":"Low"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\">
        <div class=\\"flex items-center gap-x-1 uppercase bg-ink-200 rounded-full px-1.5 py-1 cursor\\">
          <zicon-stub icon=\\"ellipsis-small\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"-mx-0.5\\"></zicon-stub> <span class=\\"text-xxs text-vanilla-400 tracking-wide font-medium\\">

  </span>
        </div>
        <div class=\\"flex items-center gap-x-1.5\\">
          <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
        </div>
        <div class=\\"flex items-center space-x-1.5\\">
          <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
        </div>
      </div>
    </div>
    <div class=\\"hidden w-full md:block md:w-1/5\\"></div>
  </div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":false,"priority":"Medium"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\">
        <div class=\\"flex items-center gap-x-1 uppercase bg-ink-200 rounded-full px-1.5 py-1 cursor\\">
          <zicon-stub icon=\\"ellipsis-small\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"-mx-0.5\\"></zicon-stub> <span class=\\"text-xxs text-vanilla-400 tracking-wide font-medium\\">

  </span>
        </div>
        <div class=\\"flex items-center gap-x-1.5\\">
          <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
        </div>
        <div class=\\"flex items-center space-x-1.5\\">
          <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
        </div>
      </div>
    </div>
    <div class=\\"hidden w-full md:block md:w-1/5\\"></div>
  </div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":false,"priority":"noop"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\">
        <div class=\\"flex items-center gap-x-1 uppercase bg-ink-200 rounded-full px-1.5 py-1 cursor\\">
          <zicon-stub icon=\\"ellipsis-small\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"-mx-0.5\\"></zicon-stub> <span class=\\"text-xxs text-vanilla-400 tracking-wide font-medium\\">
    no priority
  </span>
        </div>
        <div class=\\"flex items-center gap-x-1.5\\">
          <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
        </div>
        <div class=\\"flex items-center space-x-1.5\\">
          <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
        </div>
      </div>
    </div>
    <div class=\\"hidden w-full md:block md:w-1/5\\"></div>
  </div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":true,"priority":"High"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\"><span class=\\"relative z-menu\\"><div class=\\"leading-none\\"><button class=\\"flex items-center gap-x-1 bg-ink-200 rounded-full cursor-pointer focus:outline-none py-1 pl-1.5 pr-1\\"><!----> <span class=\\"text-vanilla-100 text-xxs font-medium tracking-wide uppercase\\">
          Set priority
        </span>
        <zicon-stub icon=\\"chevron-down\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"transition-all duration-300 transform -ml-0.5 rotate-0\\"></zicon-stub></button></div>
      <transition-stub enterclass=\\"sm:opacity-0 sm:scale-75\\" leaveclass=\\"sm:opacity-100 sm:scale-100\\" entertoclass=\\"sm:opacity-100 sm:scale-100\\" leavetoclass=\\"sm:opacity-0 sm:scale-75\\" enteractiveclass=\\"animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad\\" leaveactiveclass=\\"animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad\\">
        <!---->
      </transition-stub></span>
      <div class=\\"flex items-center gap-x-1.5\\">
        <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
      </div>
      <div class=\\"flex items-center space-x-1.5\\">
        <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
      </div>
    </div>
  </div>
  <div class=\\"hidden w-full md:block md:w-1/5\\">
    <div class=\\"flex justify-end mt-1 h-full\\">
      <zbutton-stub color=\\"cherry\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"minus-circle\\" data-testid=\\"unset-issue-priority\\" data-original-title=\\"null\\" class=\\" has-tooltip\\"></zbutton-stub>
    </div>
  </div>
</div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":true,"priority":"Low"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\"><span class=\\"relative z-menu\\"><div class=\\"leading-none\\"><button class=\\"flex items-center gap-x-1 bg-ink-200 rounded-full cursor-pointer focus:outline-none py-1 pl-1.5 pr-1\\"><!----> <span class=\\"text-vanilla-100 text-xxs font-medium tracking-wide uppercase\\">
          Set priority
        </span>
        <zicon-stub icon=\\"chevron-down\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"transition-all duration-300 transform -ml-0.5 rotate-0\\"></zicon-stub></button></div>
      <transition-stub enterclass=\\"sm:opacity-0 sm:scale-75\\" leaveclass=\\"sm:opacity-100 sm:scale-100\\" entertoclass=\\"sm:opacity-100 sm:scale-100\\" leavetoclass=\\"sm:opacity-0 sm:scale-75\\" enteractiveclass=\\"animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad\\" leaveactiveclass=\\"animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad\\">
        <!---->
      </transition-stub></span>
      <div class=\\"flex items-center gap-x-1.5\\">
        <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
      </div>
      <div class=\\"flex items-center space-x-1.5\\">
        <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
      </div>
    </div>
  </div>
  <div class=\\"hidden w-full md:block md:w-1/5\\">
    <div class=\\"flex justify-end mt-1 h-full\\">
      <zbutton-stub color=\\"cherry\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"minus-circle\\" data-testid=\\"unset-issue-priority\\" data-original-title=\\"null\\" class=\\" has-tooltip\\"></zbutton-stub>
    </div>
  </div>
</div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":true,"priority":"Medium"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\"><span class=\\"relative z-menu\\"><div class=\\"leading-none\\"><button class=\\"flex items-center gap-x-1 bg-ink-200 rounded-full cursor-pointer focus:outline-none py-1 pl-1.5 pr-1\\"><!----> <span class=\\"text-vanilla-100 text-xxs font-medium tracking-wide uppercase\\">
          Set priority
        </span>
        <zicon-stub icon=\\"chevron-down\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"transition-all duration-300 transform -ml-0.5 rotate-0\\"></zicon-stub></button></div>
      <transition-stub enterclass=\\"sm:opacity-0 sm:scale-75\\" leaveclass=\\"sm:opacity-100 sm:scale-100\\" entertoclass=\\"sm:opacity-100 sm:scale-100\\" leavetoclass=\\"sm:opacity-0 sm:scale-75\\" enteractiveclass=\\"animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad\\" leaveactiveclass=\\"animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad\\">
        <!---->
      </transition-stub></span>
      <div class=\\"flex items-center gap-x-1.5\\">
        <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
      </div>
      <div class=\\"flex items-center space-x-1.5\\">
        <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
      </div>
    </div>
  </div>
  <div class=\\"hidden w-full md:block md:w-1/5\\">
    <div class=\\"flex justify-end mt-1 h-full\\">
      <zbutton-stub color=\\"cherry\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"minus-circle\\" data-testid=\\"unset-issue-priority\\" data-original-title=\\"null\\" class=\\" has-tooltip\\"></zbutton-stub>
    </div>
  </div>
</div>
</div>"
`;

exports[`renders IssuePriorityCard with all prop options: {"title":"Assert statement used outside of tests","shortcode":"BAN-B101","issueType":"security","analyzer":{"name":"Python","logo":"analyzer_logos/python.svg","shortcode":"python"},"canChangePriority":true,"priority":"noop"} 1`] = `
"<div to=\\"\\" class=\\"block group\\">
  <!---->
  <div class=\\"flex flex-wrap flex-1 w-full\\">
    <div class=\\"flex flex-col space-y-2 text-sm border-ink-300 text-vanilla-400 justify-evenly w-full md:w-4/5 py-3\\">
      <div class=\\"flex items-center space-x-2 text-base font-semibold sm:text-lg text-vanilla-200\\">
        <p class=\\"text-sm font-semibold text-vanilla-300\\"><span>Assert statement used outside of tests</span> <span class=\\"ml-1 text-xs font-normal text-vanilla-400 whitespace-nowrap\\">BAN-B101
      </span></p>
      </div>
      <div class=\\"flex flex-wrap text-xs gap-x-4\\"><span class=\\"relative z-menu\\"><div class=\\"leading-none\\"><button class=\\"flex items-center gap-x-1 bg-ink-200 rounded-full cursor-pointer focus:outline-none py-1 pl-1.5 pr-1\\"><zicon-stub icon=\\"ellipsis-small\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"-mx-0.5\\"></zicon-stub> <span class=\\"text-vanilla-100 text-xxs font-medium tracking-wide uppercase\\">
          no priority
        </span>
        <zicon-stub icon=\\"chevron-down\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"transition-all duration-300 transform -ml-0.5 rotate-0\\"></zicon-stub></button></div>
      <transition-stub enterclass=\\"sm:opacity-0 sm:scale-75\\" leaveclass=\\"sm:opacity-100 sm:scale-100\\" entertoclass=\\"sm:opacity-100 sm:scale-100\\" leavetoclass=\\"sm:opacity-0 sm:scale-75\\" enteractiveclass=\\"animate-slide-bottom-enter-active sm:animate-none sm:transition-all sm:duration-75 sm:ease-out-quad\\" leaveactiveclass=\\"animate-slide-bottom-leave-active sm:animate-none sm:transition-all sm:duration-150 sm:ease-in-quad\\">
        <!---->
      </transition-stub></span>
      <div class=\\"flex items-center gap-x-1.5\\">
        <analyzerlogo-stub analyzerlogo=\\"\\" name=\\"Python\\" shortcode=\\"python\\" size=\\"small\\" hidetooltip=\\"true\\" logo=\\"analyzer_logos/python.svg\\"></analyzerlogo-stub> <span class=\\"text-sm tracking-wide capitalize text-vanilla-400\\">
          Python
        </span>
      </div>
      <div class=\\"flex items-center space-x-1.5\\">
        <zicon-stub icon=\\"security\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub> <span class=\\"text-sm text-vanilla-400 tracking-wide\\">Security</span>
      </div>
    </div>
  </div>
  <div class=\\"hidden w-full md:block md:w-1/5\\">
    <div class=\\"flex justify-end mt-1 h-full\\">
      <zbutton-stub color=\\"cherry\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"minus-circle\\" data-testid=\\"unset-issue-priority\\" data-original-title=\\"null\\" class=\\" has-tooltip\\"></zbutton-stub>
    </div>
  </div>
</div>
</div>"
`;
